### PR TITLE
Misc small patches

### PR DESCRIPTION
**Pull request by dequis**
* purple: cleanup, remove one usage of static local_bee
* storage_xml: replace hackish code with a GHashTableIter
* Don't hard-fail when building docs and dependencies are missing
* twitter_parse_id function, with better error handling than sscanf()

### CODE REVIEW

**Comment by dequis** [comment]
Added a few extra commits, coverity scan integration in .travis.yml (push to the coverity_scan branch to trigger it), and two commits fixing the lowest-hanging-fruit bugs reported by coverity.

File: doc/Makefile [hunk 1]
```diff
@@ -5,13 +5,13 @@ endif
 
 all:
 	# Only build the docs if this is a git tree
-	test ! '(' -d ../.git -o -d ../.bzr ')' || $(MAKE) -C user-guide
+	-test ! '(' -d ../.git -o -d ../.bzr ')' || $(MAKE) -C user-guide
```
Comment (jelmer) [inline doc/Makefile]: This change probably means that the travis build will no longer catch breakage in the documentation build. Even if it does, other package builds can now accidentally skip the docs if this is the case.

Wouldn't a configure option (--doc=0) be better for this?

**Comment by dequis** [comment]
Added --doc= configure option as suggested by jelmer, also improved the configure time checks for xmlto/xsltproc/a2x, now those aren't hard dependencies and the user is warned if they are missing.

I'd like to replace those deps with something entirely different at some point, like a python script or something. And, you know, I could even go ahead and write it in perl, which we already depend on (it's used in `doc/user-guide/Makefile` as a sed-on-steroids), so we don't add python as a dep.

**Comment by dequis** [comment]
\*beep*

**Comment by jelmer** [comment]
Doesn't look unreasonable to me.

**Comment by dequis** [comment]
ha ha merged it anyway despite your not 100% certain phrasing

**Pull request merged**